src/Exercise06AI.ipynb

```diff
@@ -71,51 +71,66 @@
     "\n",
-    "Bisher war `result` ein Tupel der Form $$result = (evaluation, move)$$\n",
-    "Um Alpha-Beta-Pruning memoisieren zu können, wird nun nicht mehr `result` im Cache gespeichert, sondern ein Tripel der Form $$ value = (flag, evaluation, move) $$ mit $$ flag \\in \\{ \\; \\text{'≤'}, \\text{'='}, \\text{'≥'} \\; \\} $$\n",
+    "### Memoisierung bei Alpha-Beta-Pruning\n",
     "\n",
-    "wobei gilt $$flag = \\begin{cases}\n",
-    "\\text{'≤'} \\; \\text{wenn} \\; evaluation \\le alpha\\\\\n",
-    "\\text{'='} \\; \\text{wenn} \\; evaluation < beta\\\\\n",
-    "\\text{'≥'} \\; \\text{sonst}\n",
-    "\\end{cases}$$\n",
+    "Prinzipiell wäre es möglich, die Memoisierung auch mit vorhandenem Alpha-Beta-Pruning genau so durchzuführen wie oben beschrieben. In diesem Fall müssten die Werte für Alpha und Beta als Identifikator mit aufgenommen werden. Da sich diese Werte aber sehr oft ändern, gäbe es nur sehr selten übereinstimmungen innerhalb der Suche und der Geschwindigkeitsvorteil wäre nur sehr klein [2]. Stattdessen wird ein anderer Ansatz gewählt [2]:\n",
     "\n",
-    "Das Speichern dieses Tupels wird von der Funktion `store_in_cache` übernommen. Diese bekommt als Argumente den Cache-Schlüssel des aktuellen Boards (`key`), das berechnete Minimax-Ergebnis als Tupel (`result`), die Grenzwerte $alpha$ und $beta$ und speichert das Ergebnis nach dem oben gegebenen Verfahren im Cache ab.\n",
+    "Anstatt nur das Ergebnis einer Berechnung zu speichern wird nun zusätzlich ein Erkennunszeichen $\\texttt{flag} \\in \\{ `\\texttt{≤}`, `\\texttt{=}`, `\\texttt{≥}` \\} $ hinzugefügt. Für ein berechnetes Ergebnis $v, m = \\textrm{minimax}(s, d, \\alpha, \\beta)$ wird also das Tripel $(\\texttt{flag}, v, m)$ im Cache abgelegt. Hierbei ist $v \\in \\mathbb{Z}$ die ermittelte Bewertung der Suche, $m$ der berechnete beste Zug, $\\textrm{minimax}$ die nicht-memoisierte Suchfunktion mit Alpha-Beta-Pruning, $s$ der aktuelle Spielzustand (Board), $d \\in \\mathbb{N}$ die gewählte Suchtiefe und $\\alpha, \\beta \\in \\mathbb{Z}$ sind die Grenzen des Alpha-Beta-Prunings. Der Identifikator für den Cache bleibt weiterhin wie oben beschrieben, also ohne die Werte Alpha und Beta.\n",
```

Comment:
I'd suggest to consistently use textrm for variable names inside math blocks. We should also fix this in the lines below for the `Cache` variable
```suggestion
    "Anstatt nur das Ergebnis einer Berechnung zu speichern wird nun zusätzlich ein Erkennunszeichen $\\textrm{flag} \\in \\{ `\\texttt{≤}`, `\\texttt{=}`, `\\texttt{≥}` \\} $ hinzugefügt. Für ein berechnetes Ergebnis $v, m = \\textrm{minimax}(s, d, \\alpha, \\beta)$ wird also das Tripel $(\\textrm{flag}, v, m)$ im Cache abgelegt. Hierbei ist $v \\in \\mathbb{Z}$ die ermittelte Bewertung der Suche, $m$ der berechnete beste Zug, $\\textrm{minimax}$ die nicht-memoisierte Suchfunktion mit Alpha-Beta-Pruning, $s$ der aktuelle Spielzustand (Board), $d \\in \\mathbb{N}$ die gewählte Suchtiefe und $\\alpha, \\beta \\in \\mathbb{Z}$ sind die Grenzen des Alpha-Beta-Prunings. Der Identifikator für den Cache bleibt weiterhin wie oben beschrieben, also ohne die Werte Alpha und Beta.\n",
```

Reply:
Fixed in a56af57cdd0761211a8a79f3e7c33274d008d962 ✅